testsuite/MDAnalysisTests/test_persistence.py

```diff
@@ -36,10 +36,8 @@
 import shutil
 import warnings
 
-"""These tests need to be reworked for the new 363 topoloy system
 
-"""
-class SkipAtomGroupPickle(object):
+class TestAtomGroupPickle(object):
```

Comment:
I'd say it's still an anchor if you specify anchor_name, it's just that you've controlled the hash representation of the Universe.  I replaced the two weaksets with a single weak dictionary, there's no chance that the keys will collide (unless people start calling their anchors -3590150986908 -_-)